honestcash-v2/src/app/user/store/user.effects.ts

```diff
@@ -0,0 +1,47 @@
+import {Injectable} from '@angular/core';
+import {Actions, Effect, ofType} from '@ngrx/effects';
+import {Observable} from 'rxjs';
+import {map, switchMap, tap} from 'rxjs/operators';
+
+import {UserActionTypes, UserSetup} from './user.actions';
+import {LoginSuccessResponse, SignupSuccessResponse} from '../../auth/models/authentication';
+import {UserService} from '../services/user.service';
+
+@Injectable()
+export class UserEffects {
+
+  constructor(
+    private actions: Actions,
+    private userService: UserService,
+  ) {
+  }
+
+  @Effect({dispatch: false})
+  public UserSetup: Observable<any> = this.actions.pipe(
+    ofType(UserActionTypes.USER_SETUP),
+    map((action: UserSetup) => action.payload),
+    tap((payload?: LoginSuccessResponse | SignupSuccessResponse) => {
+      if (payload) {
+        return this.userService.init(payload.token, payload.user);
+      }
+      this.userService.init();
```

Comment:
if there is no payload (meaning it is either MainLoad or SignUpSuccess) init checks localstorage (if code reaches here, it will already be checked for existence of localstorage keys for both token and wallet by then via MainLoad effect)